catballotbox.go

```diff
@@ -39,12 +53,13 @@ func NewCatBallotBox(size int) *CatBallotBox {
 //Vote registers a vote that case "casei" should be predicted to be the
 //category "pred".
 func (bb *CatBallotBox) Vote(casei int, pred string, weight float64) {
+
 	predn := bb.CatToNum(pred)
 	bb.Box[casei].Mutex.Lock()
-	if _, ok := bb.Box[casei].Map[predn]; !ok {
-		bb.Box[casei].Map[predn] = 0
+	for len(bb.Box[casei].Cat) <= predn {
```

Comment:
I think `predn` is always sequential, so could the `for` be replaced with an `if`?

Reply:
it panics with an `if` instead of a `for`, panic: runtime error: index out of range, on TestThreeTargets, for example.